### PR TITLE
fix(ui): Tiny modal DAG tweaks. Fixes #4039

### LINKED ISSUE #4039: Tooltips should also be supplied for ellipsized text
# Summary

Tooltips (introduced in #767) should extend to the text below icons

# Use Cases

Visit a workflow and see:
image

If you hover over the ✅ , you'll get a tooltip.
If you hover over the ellipsized text below, you don't

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [ ] ~~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~~
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

Fixes #4039
Related to #3595 (#3967)

Screenshot 2020-09-15 at 15 28 52
Screenshot 2020-09-15 at 15 30 43
Screenshot 2020-09-15 at 16 39 55
Screenshot 2020-09-15 at 15 28 18

### CODE REVIEW

File: ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx [hunk 1]
```diff
@@ -40,12 +40,8 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         return 32 / this.scale;
     }
 
-    private get hgap() {
-        return this.nodeSize * 2;
-    }
-
-    private get vgap() {
-        return this.nodeSize;
+    private get gap() {
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:43]: using a single gap value makes layout more predictable

File: ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx [hunk 2]
```diff
@@ -560,6 +552,9 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         }
 
         const node = this.props.nodes[id];
+        if (!node) {
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:555]: fixes crashing issue

File: ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx [hunk 3]
```diff
@@ -568,6 +563,12 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
     }
 
     private layoutGraph(nodes: string[], edges: Edge[]) {
+        const hash = {scale: this.scale, nodeCount: nodes.length, nodesToDisplay: this.state.nodesToDisplay};
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:566]: move this check here so both graphs benefit

File: ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx [hunk 4]
```diff
@@ -397,10 +394,10 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         const graph = new dagre.graphlib.Graph();
 
         graph.setGraph({
-            edgesep: 20 / this.scale,
-            nodesep: 50 / this.scale,
+            edgesep: this.gap,
+            nodesep: this.gap,
             rankdir: this.state.horizontal ? 'LR' : 'TB',
-            ranksep: 50 / this.scale
+            ranksep: this.gap
```
Comment (simster7) [inline ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:400]: Does this change how the graph looks? Could you add a screenshot?

**Review by simster7 — CHANGES_REQUESTED** [review]

**Comment by simster7** [comment]
Sorry I know this is still in draft mode... jumped the gun on the review

File: ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx (hunk 4 above)
Comment (alexec) [reply ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:400]: this makes it more compact - moving from 50px sep to 32px sep (default zoom)

File: ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx (hunk 2 above)
Comment (alexec) [reply ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:555]: I don't think this will conflict looking at #4028?

Comment (simster7) [reply ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:555]: This code will be unreachable after #4028. Also this fix is not exhaustive as there are other places in the file where null nodes may be returned under the right conditions. #4028 fixes them exhaustively.

Comment (alexec) [reply ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx:555]: deleted

**Review by simster7 — APPROVED** [review]

**Pull request merged**